lapis/file_provider.py

```diff
@@ -0,0 +1,37 @@
+from lapis.storage import Storage
+
+
+class FileProvider(object):
+
+    __slots__ = ("storages",)
+
+    def __init__(self):
+        self.storages = dict()
+
+    def add_storage_element(self, storage_element: Storage):
+        try:
+            self.storages[storage_element.sitename].append(storage_element)
+        except KeyError:
+            self.storages[storage_element.sitename] = [storage_element]
+
+    def provides_all_files(self, job):
```

Comment:
Please consider providing a method that determines a score and not a check for strictly all files. Meaning: consider renaming the method. The implementation can stay as is as long as you don't need something more advanced.